ibis/sql/mssql/compiler.py

```diff
@@ -0,0 +1,207 @@
+import pyodbc
+import sqlalchemy as sa
+import sqlalchemy.dialects.mssql as mssql
+
+
+import ibis.sql.alchemy as alch
+import ibis.expr.datatypes as dt
+import ibis.expr.operations as ops
+import ibis.common.exceptions as com
+
+# used for literal translate
+from ibis.sql.alchemy import unary, fixed_arity
+
+
+def raise_unsupported_op_error(translator, expr, *args):
+    msg = "SQLServer backend doesn't support {} operation!"
+    op = expr.op()
+    raise com.UnsupportedOperationError(msg.format(type(op)))
+
+
+# Aggregation
+# coppied from postgresql compiler
+# support for of bit columns in aggregate methods
+def _reduction(func_name, cast_type='int32'):
+    def reduction_compiler(t, expr):
+        arg, where = expr.op().args
+
+        if arg.type().equals(dt.boolean):
+            arg = arg.cast(cast_type)
+
+        func = getattr(sa.func, func_name)
+
+        if where is not None:
+            arg = where.ifelse(arg, None)
+        return func(t.translate(arg))
+
+    return reduction_compiler
+
+
+# String
+# TODO: substr and find are copied from SQLite, we should really have a
```

Comment:
isn't ibis/sql/* the place for this?

happy to refactor this later, though might be useful to do a pre-cursor where you factor out from the various sqlalchemy based backends

Reply:
I removed the comments since this method actually cannot use the sqlite method due to the sql function having a different name. But otherwise this method is very similar. Looks like the section you are talking about is in `ibis/sql/*` were you commenting on a different file?

Reply:
after this PR it makes sense to refactor the sql backends to share common code - per backend names can easily be parameter used - copying code is not great - but as a follow up is ok

if u can create an issue